### PR TITLE
Init translation of the tutorial of Cascade layers

### PR DESCRIPTION
**Pull request by hanyujie2002**
### Description

Init the translation of the tutorial of Cascade layers

Translate all contents of [cascade layer](https://github.com/mdn/content/blob/main/files/en-us/learn/css/building_blocks/cascade_layers/index.md?plain=1)
### Motivation

The readers who use Chinese language could read the article more easily

### Additional details

- There are two terms with subtle difference which I think should be distinguished in the Chinese translation: "specificity" and "precedence". In my opinion, the "specificity" is directly related to the CSS selector, while the later not. So I translate the "specificity" into "优先级" and translate "precedence" into "优先权".
- There are some inner links which is invalid in the original article(for example [this](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_layers#nested-layers)), to respect the original content, I don't fix that in the translation
- The "cascade" in the former articles are translated into "层叠", not "级联", so I translate "cascade layer" into "层叠层"

### Related issues and pull requests

None

### CODE REVIEW

**Comment by hanyujie2002** [comment]
Why that can't pass the check? Please help me @yin1999

**Comment by jasonren0403** [comment]
> Why that can't pass the check? Please help me @yin1999

Hi @hanyujie2002 , it was the yaml metadata format problem, I allowed myself for a fix, so this should work now 👍

**Comment by hanyujie2002** [comment]
> > Why that can't pass the check? Please help me @yin1999
> 
> Hi @hanyujie2002 , it was the yaml metadata format problem, I allowed myself for a fix, so this should work now 👍

Thank you very much ヾ(≧▽≦*)o

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
+---
+title: 层叠层
+slug: Learn/CSS/Building_blocks/Cascade_layers
+---
+
+{{LearnSidebar}}{{PreviousMenuNext("Learn/CSS/Building_blocks/Cascade_and_inheritance", "Learn/CSS/Building_blocks/The_box_model", "Learn/CSS/Building_blocks")}}
+
+这一课的目的是向你介绍[层叠层](/zh-CN/docs/Web/CSS/@layer)，这是一个更高级的特性，它建立在 [CSS 层叠](/zh-CN/docs/Web/CSS/Cascade) 和 [CSS 优先级](/zh-CN/docs/Web/CSS/Specificity) 的基本概念之上。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:8]: remove the spaces

```suggestion
这一课的目的是向你介绍[层叠层](/zh-CN/docs/Web/CSS/@layer)，这是一个更高级的特性，它建立在 [CSS 层叠](/zh-CN/docs/Web/CSS/Cascade)和 [CSS 优先级](/zh-CN/docs/Web/CSS/Specificity)的基本概念之上。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+如果你是 CSS 的新手，刚开始可能会觉得这部分的内容与本课程的其他部分相比不太相关，而且有些学术化。然而，了解层叠层的基本知识对于你在项目中遇到它们时会非常有帮助。随着你对 CSS 的不断使用，理解层叠层以及如何充分利用它们的功能将能够避免在处理来自不同团队、插件和开发人员的 CSS 代码库时遇到的很多问题。
+
+当你使用来自多个来源的 CSS 时，层叠层最为相关，因为这时会有冲突的 CSS 选择器和竞争的优先级，或者当你考虑使用 `!important` 时。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:12]: Missing the link

```suggestion
当你使用来自多个来源的 CSS 时，层叠层最为相关，因为这时会有冲突的 CSS 选择器和竞争的优先级，或者当你考虑使用 [`!important`](/zh-CN/docs/Web/CSS/important) 时。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+    <tr>
+      <th scope="row">目标：</th>
+      <td>
+        学习层叠层是如何工作的。
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+应用于元素的每个 CSS 属性，只能有一个值。你可以通过在浏览器的开发者部件中检查元素来查看应用于元素的所有属性值。部件的“样式”面板显示了应用在被检查元素上的所有属性值，以及匹配的选择器和 CSS 源文件。具有优先权的来源的选择器将其值应用于匹配的元素。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:31]:
```suggestion
应用于元素的每个 CSS 属性，只能有一个值。你可以通过在浏览器的开发者工具中检查元素来查看应用于元素的所有属性值。工具的“样式”面板显示了应用在被检查元素上的所有属性值，以及匹配的选择器和 CSS 源文件。具有优先权的来源的选择器将其值应用于匹配的元素。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+除了应用的样式外，“样式”面板还显示了与选定元素匹配但由于层叠、优先级或源顺序而未应用的被划掉的值。划掉的样式可能来自具有相同优先权 (precedence) 来源但具有较低优先级 (specificity)，或者具有同优先权来源和优先级，但在代码库中较早出现。对于任何应用的属性值，可能有来自许多不同来源的几个声明被划掉。如果你看到一个划掉的样式，它有一个具有更高优先级的选择器，这意味着该值在来源或重要性上不足。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:33]: Use full width punctuation

```suggestion
除了应用的样式外，“样式”面板还显示了与选定元素匹配但由于层叠、优先级或源顺序而未应用的被划掉的值。划掉的样式可能来自具有相同优先权（precedence）来源但具有较低优先级（specificity），或者具有同优先权来源和优先级，但在代码库中较早出现。对于任何应用的属性值，可能有来自许多不同来源的几个声明被划掉。如果你看到一个划掉的样式，它有一个具有更高优先级的选择器，这意味着该值在来源或重要性上不足。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+## CSS 层叠概念的回顾
+
+CSS 中的 C 代表“层叠”。这是样式层叠在一起的方法。用户代理经过几个非常明确定义的步骤来确定每个元素的每个属性的值。我们将在这里简要列出这些步骤，然后深入研究第 4 步：层叠层，就是你来到这里学习的内容：
+
+1. **相关声明：** 找到所有具有匹配每个元素的选择器的声明代码块。
+2. **重要性：** 根据规则是普通还是 important 对规则进行排序。重要的样式是指设置了 [`!important`](/zh-CN/docs/Web/CSS/important) 标志的样式。
+3. **来源：** 在两个按重要性划分的分组内，按作者、用户或用户代理这几个来源对规则进行排序。
+4. **层：** 在六个按重要性和来源划分的分组内，按层叠层进行排序。普通声明的层顺序是从第一个创建到最后一个，然后是未分层的普通样式。对于重要的样式，这个顺序是反转的，有层次的重要样式优先权最低。
+5. **优先级：** 对于同一层中具有优先级的竞争样式，按 [优先级](/zh-CN/docs/Web/CSS/Specificity) 对声明进行排序。
+6. **出现顺序：** 当两个起源层的优先级相同的选择器具有相同的优先级时，最后声明的选择器的属性值获胜。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:48]:
```suggestion
1. **相关声明**：找到所有具有匹配每个元素的选择器的声明代码块。
2. **重要性**：根据规则是普通还是重要对规则进行排序。重要的样式是指设置了 [`!important`](/zh-CN/docs/Web/CSS/important) 标志的样式。
3. **来源**：在两个按重要性划分的分组内，按作者、用户或用户代理这几个来源对规则进行排序。
4. **层**：在六个按重要性和来源划分的分组内，按层叠层进行排序。普通声明的层顺序是从第一个创建到最后一个，然后是未分层的普通样式。对于重要的样式，这个顺序是反转的，有层次的重要样式优先权最低。
5. **优先级**：对于同一层中具有优先级的竞争样式，按 [优先级](/zh-CN/docs/Web/CSS/Specificity) 对声明进行排序。
6. **出现顺序**：当两个起源层的优先级相同的选择器具有相同的优先级时，最后声明的选择器的属性值获胜。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+对于每一步，只有“仍在运行”的声明才会进入下一轮"竞争"。如果只有一个声明在运行，那么它就“赢了”, 后续的步骤就没有意义了。
+
+### 来源和层叠
+
+有三种 [层叠来源类型](/zh-CN/docs/Web/CSS/Cascade#origin_types)：用户代理样式表、用户样式表和作者样式表。浏览器根据来源和重要性将每个声明分为六个来源分组。有八个优先级：六个来源分组、正在过渡的属性和正在动画的属性。优先级的顺序是从具有最低优先级的普通用户代理样式，到当前应用的动画中的样式，到具有最高优先级的重要用户代理样式，再到正在过渡的样式：
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:54]:
```suggestion
有三种[层叠来源类型](/zh-CN/docs/Web/CSS/Cascade#origin_types)：用户代理样式表、用户样式表和作者样式表。浏览器根据来源和重要性将每个声明分为六个来源分组。有八个优先级：六个来源分组、正在过渡的属性和正在动画的属性。优先级的顺序是从具有最低优先级的普通用户代理样式，到当前应用的动画中的样式，到具有最高优先级的重要用户代理样式，再到正在过渡的样式：
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+>
+> —[层叠与继承规范 ↗](https://www.w3.org/TR/css-cascade-5/#layering)。
+
+## 创建层叠层
+
+可以使用以下任一方法创建层叠层：
+
+- 使用`@layer` 声明规则，使用`@layer` 后跟一个或多个层的名称来声明层。这将创建一个没有分配任何样式的命名层。
+- 使用`@layer` 块规则，在块中的所有样式都将添加到一个命名或未命名的层中。
+- 使用具有`layer` 关键字或 `layer()`函数的[`@import`](/zh-CN/docs/Web/CSS/@import) 规则，将导入文件的内容分配到该层中。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:131]:
```suggestion
- 使用 `@layer` 声明规则，使用 `@layer` 后跟一个或多个层的名称来声明层。这将创建一个没有分配任何样式的命名层。
- 使用 `@layer` 块规则，在块中的所有样式都将添加到一个命名或未命名的层中。
- 使用具有 `layer` 关键字或 `layer()` 函数的 [`@import`](/zh-CN/docs/Web/CSS/@import) 规则，将导入文件的内容分配到该层中。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+
+在尚未初始化具有相同名称的层的情况下，这三种方法中的任何一种都会创建一个层。如果在 `@layer` 规则或带有 `layer()`的`@import` 中没有提供层名称，则将创建一个新的匿名层。
+
+> \*\*注意：\*\*层的优先顺序是它们创建的顺序。不在层中的样式，或者称为“未分层样式”，会层叠到最终的隐式标签中。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:135]: Please check: https://developer.mozilla.org/zh-CN/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#%E5%A4%87%E6%B3%A8%E3%80%81%E8%AD%A6%E5%91%8A%E5%92%8C%E6%A0%87%E6%B3%A8

```suggestion
> **备注：** 层的优先顺序是它们创建的顺序。不在层中的样式，或者称为“未分层样式”，会层叠到最终的隐式标签中。
```

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+}
+```
+
+在上面的 CSS 中，我们创建了五个层：`layout`、`<匿名 (01)>`、`theme`、`utilities` 和 `<匿名 (02)>` - 按这个顺序 - 第六个隐含的未分层样式层包含在`body`样式块中。层的顺序是层的创建顺序，未分层样式的隐含层总是最后的。一旦创建了层之后就无法改变层的顺序。
+
+我们将一些样式分配给名为 `layout` 的层。如果指定的命名层不存在，则在`@layer` 规则中指定名称（无论是否向层分配样式）都会创建该层；这会将该层添加到现有层名称系列的末尾。如果指定的命名层已经存在，该命名块内的所有样式都会附加到之前存在的层中的样式 - 通过重用现有的层名称指定样式不会创建新层。
+
+通过在不命名层的情况下将样式分配给层来创建匿名层。只能在创建时向未命名的层添加样式。
+
+> **注意：** 后续使用不带层名的`@layer` 会创建额外的未命名层；它不会将样式附加到先前存在的未命名层。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:210]: same

**Review by yin1999 — COMMENTED** [review]
Hey @hanyujie2002. I've listed some common problem. Could you address all of them and make a fix to this PR.

You could check the [translation guide](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md) for more information.

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+要了解层叠层，必须很好地理解 CSS 层叠。下面的小节提供了层叠的重要概念的快速回顾。
+
+## CSS 层叠概念的回顾
+
+CSS 中的 C 代表“层叠”。这是样式层叠在一起的方法。用户代理经过几个非常明确定义的步骤来确定每个元素的每个属性的值。我们将在这里简要列出这些步骤，然后深入研究第 4 步：层叠层，就是你来到这里学习的内容：
+
+1. **相关声明**：找到所有具有匹配每个元素的选择器的声明代码块。
+2. **重要性**：根据规则是普通还是重要对规则进行排序。重要的样式是指设置了 [`!important`](/zh-CN/docs/Web/CSS/important) 标志的样式。
+3. **来源**：在两个按重要性划分的分组内，按作者、用户或用户代理这几个来源对规则进行排序。
+4. **层**：在六个按重要性和来源划分的分组内，按层叠层进行排序。普通声明的层顺序是从第一个创建到最后一个，然后是未分层的普通样式。对于重要的样式，这个顺序是反转的，有层次的重要样式优先权最低。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md]: 未分层的样式始终具有最低的优先权

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+5. 作者重要样式
+6. 用户重要样式
+7. 用户代理重要样式
+8. 正在过渡的样式
+
+“用户代理”指的是浏览器。“用户”指的是是网站访问者。“作者”指的是你，开发者。用 {{HTMLElement('style')}} 元素直接在元素上声明的样式是作者样式。不包括动画和过渡样式，用户代理普通样式具有最低优先权；用户代理重要样式具有最高优先权。
+
+### 来源和优先级
+
+对于每个属性，“获胜”的声明是来自基于权重（普通或重要）具有优先权的来源的声明。暂时忽略层，来自具有最高优先权的来源的值将被应用。如果获胜来源对于一个元素有多个属性声明，那么这些竞争属性值的选择器的优先级将被比较。不同来源之间的选择器从不比较优先级。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:69]: 请从此处开始继续检查后文的翻译，并在准备完毕后再通知我进行审查

**Review by yin1999 — COMMENTED** [review]
请再检查一下后文的翻译，感谢 🙏

**Review by yin1999 — COMMENTED** [review]
最后再考虑一个问题：是否要将“named layer”翻译成“具名层”，在我们其他的 CSS 文档中采用的是这个翻译。当然，当前文档中不是所有的“named”都可以翻译为“具名”请视情况而定

File: files/zh-cn/learn/css/building_blocks/cascade_layers/index.md [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+
+### 嵌套层的优点
+
+嵌套层的能力使团队可以创建层叠层，而不用担心其他团队会将它们导入到一个层中。同样，嵌套使你可以将第三方样式表导入到一个层中，而不用担心该样式表本身是否具有层。因为层可以嵌套，所以你不必担心外部和内部样式表之间会有冲突的层名称。
+
+### 创建嵌套层叠层
+
+嵌套层可以使用与常规层相同的方法创建。例如，可以使用点表示法，在 `@layer` 规则后跟一个或多个层名称来创建。多个点和层名称表示多重嵌套。
+
+如果你在一个块 `@layer` at 规则内嵌套另一个块 `@layer` at 规则，无论是否命名，嵌套的块都会成为一个嵌套层。类似地，当一个样式表被使用包含 `layer` 关键字或 `layer()` 函数的 `@import` 声明导入时，样式会被分配到该命名或匿名层。如果 `@import` 语句包含层，那么这些层会成为该匿名或具名层中的嵌套层。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/cascade_layers/index.md:276]:
```suggestion
如果你在一个块 `@layer` at 规则内嵌套另一个块 `@layer` at 规则，无论是否命名，嵌套的块都会成为一个嵌套层。类似地，当一个样式表被使用包含 `layer` 关键字或 `layer()` 函数的 `@import` 声明导入时，样式会被分配到该具名或匿名层。如果 `@import` 语句包含层，那么这些层会成为该匿名或具名层中的嵌套层。
```

**Review by yin1999 — APPROVED** [review]
感谢 🎉

**Pull request merged**